### PR TITLE
feat: Add POST device resource tests

### PR DESCRIPTION
**Pull request by jumpingliu**
feat #619

### CODE REVIEW

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+*** Variables ***
+${SUITE}          Core Metadata Device Profile POST Deviceresource Negative Test Cases
+${LOG_FILE_PATH}  ${WORK_DIR}/TAF/testArtifacts/logs/core-metadata-deviceprofile-post-deviceresource-negative.log
+
+*** Test Cases ***
+ErrProfileResourcePOST001 - Create device resource with duplicate resource name
+    # 2 device resource with same resource name
+    Given Generate A Device Profile Sample
+    And Should Return Status Code "201"
```
Comment (cherrycl) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot:17]: We don't have to contain this check here.

**Review by cherrycl — COMMENTED** [review]
For negative tests, not only validate the empty field, but need to contain the invalid data.
For positive tests, add tests to add single resource and multiple resource on each request.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    And Item Index All Should Contain Status Code "201" And id
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ProfileResourcePOST003 - Add multiple deviceResources on one device profile basicinfo only
+    # multiple resources > one basicinfo profile
+    Given Create A Device Profile Basicinfo Only
+    When Add multiple New resources on Device Profile Basicinfo Only
+    Then Should Return Status Code "207"
+    And Should Return Content-Type "application/json"
+    And Item Index All Should Contain Status Code "201" And id
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:38]: The response doesn't return id.
And add a check to verify the new device resources are added.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Force Tags      Skipped
+
+*** Variables ***
+${SUITE}          Core Metadata Device Profile POST Resource Positive Test Cases
+${LOG_FILE_PATH}  ${WORK_DIR}/TAF/testArtifacts/logs/core-metadata-deviceprofile-post-deviceresource-positive.log
+
+*** Test Cases ***
+ProfileResourcePOST001 - Add deviceResource on one device profile basicinfo only
+    # one resource > one basicinfo profile
+    Given Create A Device Profile Basicinfo Only
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:16]: Create a profile that already contains device resources instead.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    When Add A New resource on device profile Basicinfo Only
+    Then Should Return Status Code "207"
+    And Should Return Content-Type "application/json"
+    And Item Index All Should Contain Status Code "201" And id
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:20]: The response doesn't return id. 
And add a check to verify the new device resource is added.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ProfileResourcePOST002 - Add deviceResource on multiple device profiles basicinfo only
+    # one resource > multiple basicinfo profiles
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:24]: we don't need this testcase.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ProfileResourcePOST004 - Add multiple deviceResources on multiple device profiles
+    # multiple resources on profile basicinfo > multiple basicinfo profiles & profiles sample
+    Given Create multiple Device Profiles Basicinfo Only
+    And Create Multiple Device Profiles Sample
+    When Add multiple New resources on Device Profiles Basicinfo Only
+    Then Should Return Status Code "207"
+    And Should Return Content-Type "application/json"
+    And Item Index All Should Contain Status Code "201" And id
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:48]: The response doesn't return id.
And add a check to verify the new device resources are added to the profiles correctly.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot [hunk 7]
```diff
@@ -0,0 +1,63 @@
+*** Settings ***
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot:1]: Add Content-Type and response time checks to all testcases.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+ErrProfileResourcePOST001 - Add deviceResource with Empty device profile name
+    # Empty device profile name
+    When Add New Resource With Empty device profile name
+    Then Should Return Status Code "404"
+
+ErrProfileResourcePOST002 - Add deviceResource with Non-existent profile name
+    # Non-existent device profile name
+    Given Create A Device Profile Sample
+    When Add New Resource With Non-existent profile name
+    Then Should Return Status Code "404"
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot:22]: This case should return status code 207 and the response code inside is 404.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+ErrProfileResourcePOST003 - Add deviceResource with duplicate Resource name
+    # 2 deviceResource with same resource name
+    Given Create A Device Profile Sample
+    When Add New Resource With the Same Resource Name
+    Then Should Return Status Code "400"
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot:28]: This case should return status code 207 and the response code inside is 400.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot [hunk 10]
```diff
@@ -0,0 +1,82 @@
+*** Settings ***
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot:1]: Add `[Teardown]` to delete device profile to the testcases containing `Create A Device Profile Sample`.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot [hunk 11]
```diff
@@ -0,0 +1,44 @@
+*** Settings ***
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:1]: Add `[Teardown]` to each testcases to delete device profiles.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**